app/controllers/delivery/address_details.js

```diff
@@ -1,33 +1,45 @@
-import Ember from 'ember';
+import Ember from "ember";
 import { translationMacro as t } from "ember-i18n";
 
 export default Ember.Controller.extend({
-
   i18n: Ember.inject.service(),
   delivery: Ember.computed.alias("deliveryController.model"),
-  user: Ember.computed.alias('delivery.offer.createdBy'),
+  user: Ember.computed.alias("delivery.offer.createdBy"),
   selectedTerritory: null,
   selectedDistrict: null,
 
-  initSelectedTerritories: Ember.on('init', function() {
-    if(this.get("selectedDistrict") === null) {
-      this.set("selectedTerritory", this.get("user.address.district.territory"));
+  initSelectedTerritories: Ember.on("init", function() {
+    if (this.get("selectedDistrict") === null) {
+      this.set(
+        "selectedTerritory",
+        this.get("user.address.district.territory")
+      );
       this.set("selectedDistrict", this.get("user.address.district"));
     }
   }),
 
   territoriesPrompt: t("all"),
   destrictPrompt: t("delivery.select_district"),
 
-  territories: Ember.computed(function(){
-    return this.store.peekAll('territory');
+  territories: Ember.computed(function() {
+    return this.store.peekAll("territory");
   }),
 
-  districtsByTerritory: Ember.computed('selectedTerritory', function(){
-    if(this.selectedTerritory && this.selectedTerritory.id) {
-      return this.selectedTerritory.get('districts').sortBy('name');
+  districtsByTerritory: Ember.computed("selectedTerritory", function() {
+    if (this.selectedTerritory && this.selectedTerritory.id) {
+      return this.selectedTerritory.get("districts").sortBy("name");
     } else {
-      return this.store.peekAll('district').sortBy('name');
+      return this.store.peekAll("district").sortBy("name");
+    }
+  }),
+
+  actions: {
+    onTerritoryChange(value) {
+      this.set("selectedTerritory", value);
+      this.set(
+        "selectedDistrict",
+        this.selectedTerritory.get("districts").sortBy("name")[0]
```

Comment:
Do we have the requirement to show the first district selected?

Reply:
nope. But its fine to have this way. Else when you change the district, the territories set related to the particular district changes. And on saving, the district will have a completely unrelated territory combination : )